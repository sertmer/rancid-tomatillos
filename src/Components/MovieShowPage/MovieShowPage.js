import React, { Component } from 'react';
import './MovieShowPage.scss';
import { connect } from 'react-redux';

class MovieShowPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      rating: null
    }
  }

  render() {
    console.log(this.props)
      return (
        <main>
          <article>
            <div>
              <h2>{this.props.currentMovie.title}</h2>
              <h2>{this.props.currentMovie.average_rating}</h2>
              <h2>{this.props.currentMovie.release_date}</h2>
            </div>
            <img className='movie-backdrop' src={this.props.currentMovie.backdrop_path} alt='large image with characters from film' />
            <div>{this.props.currentMovie.user_rating}</div>
            <div className='rating-div'>
              <label>Rate This Movie</label>
              <select className='rating-selector'>
                <option value='1'>1</option>
                <option value='2'>2</option>
                <option value='3'>3</option>
                <option value='4'>4</option>
                <option value='5'>5</option>
                <option value='6'>6</option>
                <option value='7'>7</option>
                <option value='8'>8</option>
                <option value='9'>9</option>
                <option value='10'>10</option>
              </select>
              <button>Submit Rating</button>
            </div>
            <p>{this.props.currentMovie.overview}</p>
          </article>
        </main>
      )
  }
}

export const mapState = state => ({
  currentMovie: state.currentMovie
})

export default connect(mapState)(MovieShowPage)
