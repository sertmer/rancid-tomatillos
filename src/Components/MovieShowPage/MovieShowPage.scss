@import '../../scss/mixins';
@import '../../scss/variables';

article {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-top: 3.5%;
}

.movie-backdrop {
  @include rounded-corners(5px);
  height: 60vh;
  // width: 70vw;
  margin: 1.5%
}

.show-page-header {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.show-page-info {
  @include rounded-corners(5px);
  height: 60vh;
  display: flex;
  flex-direction: column;
  margin: 1.5% 1.5% 0 0;
  align-items: center;
  background: rgb(193,193,193);
  background: linear-gradient(
    0deg, 
    rgba(193,193,193,1) 0%, 
    rgba(245,245,245,1) 100%
  );
}

p {
  margin: 5%;
}

h2 {
  font-family: $accent-font;
  font-size: xx-large;
}

.rate-movie-button {
  @include rounded-corners(5px);
  background-color: $green;
  width: 40%;
  height: 40%;
  font-family: $accent-font;
  font-size: larger;
}

.rating-div {
  margin: 2.5%;
  width: 90%;
  height: 15%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}