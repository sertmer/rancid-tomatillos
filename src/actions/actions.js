export const getMovies = movies => ({
  type: 'GET_MOVIES',
  movies
});

export const currentUser = user => ({
  type: 'SET_CURRENT_USER',
  name: user.user.name,
  id: user.user.id,
  ratings: []
});

export const loggedIn = loggedIn => ({
  type: 'LOG_IN',
  loggedIn
});

export const getCurrentMovie = movie => ({
  type: 'GET_MOVIE',
  movie
});

export const getRatings = ratings => ({
  type: 'GET_RATINGS',
  ratings
});

