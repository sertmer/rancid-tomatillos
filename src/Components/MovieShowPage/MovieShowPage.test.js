import React from 'react';
import { MovieShowPage, mapState }from './MovieShowPage';
import { shallow } from 'enzyme';
import { getCurrentMovie, currentUser, getRatings } from '../../actions/actions';

describe('MovieShowPage', () => {
  let wrapper
  let mockProps = {
    currentMovie: {
      id: 1,
      title: "Jumanji: The Next Level",
      poster_path: "url.com",
      backdrop_path: "url.com",
      release_date: "2019-12-04",
      overview: "description",
      average_rating: 6
  },
    currentUser: {
      id: 1,
      name: 'Rick'
    },
    ratings: [{
      "id": 210,
      "user_id": 8,
      "movie_id": 2,
      "rating": 7,
      "created_at": "2020-01-04T20:37:36.154Z",
      "updated_at": "2020-01-04T20:37:36.154Z"
    }]
  }
    beforeEach(() => {
      wrapper = shallow(<MovieShowPage 
        currentMovie={mockProps.currentMovie}
        currentUser={mockProps.currentUser}
        ratings={mockProps.ratings}
      />)
    });
  
    it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });