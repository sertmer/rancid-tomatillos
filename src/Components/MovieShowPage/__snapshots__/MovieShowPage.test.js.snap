// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieShowPage Should match the snapshot 1`] = `ShallowWrapper {}`;
